python/lsst/meas/base/baseMeasurement.py

```diff
@@ -485,3 +485,21 @@ def doMeasurementN(self, plugin, measCat, *args, **kwds):
                 self.log.getChild(plugin.name).warning(
                     "Exception in %s.measureN on records %s-%s: %s",
                     plugin.name, measCat[0].getId(), measCat[-1].getId(), error)
+
+    @staticmethod
+    def getFootprintsFromCatalog(catalog):
```

Comment:
I wonder if this should be in AFW instead. For example, it could be added to https://github.com/lsst/afw/blob/main/python/lsst/afw/table/_source.py#L33

Reply:
I thought about that briefly, but I think this particular dict format is specific for the noise replacer code used here, and thus should be in `meas_base`.  So maybe this should be a static method of `NoiseReplacer` instead?  (Though this would then make the call in `finalizeCharacterization` less obvious).

Reply:
Yeah, that's true. I think it's fine to leave it here.